### ISSUE TITLE
Connection Graph does not show midi port aliases anymore since 0.9.1 up until 0.9.4

### ISSUE THREAD

**Issue by henning** [rncbc/qjackctl#143]
After upgrading my Ubuntu system to 21.4, the Graph view did not show the Alias names of Jack (not alsa) Midi Ports anymore, which before the update worked fine, and was very useful to really understand which ports are connected with each other. 
Now it's just "midi_capture_N".

Having a significant number of devices with man y ports altogether, it makes it impossible to manage my connections there.

At the same time, the "normal" connection view seems also not to work as expected as soon as I chose the display config that shows the port names. 

I then get the names, but connections are not shown, and trying to make a new one by 

In the mode where the aliases are also not shown, it seems to work - I see and can make new connections. But again I dont see what I connect because just numbers are shown.

I realized the version in Ubuntu is outdated(0.9.1) but getting the latest version and building the deb and installing it doesnt change this behaviour.

Is there any other way that let's me see and modify the jack midi connections with their human identifyable Alias names in qjackctl? 
I also tried the patchbay, but even though midi ports are shown there, and I can draw connections, it seems that nothing happens with these, no connections are made. 
The only documentation about it I find is the blog post from 2008, it also says it does not support Jack midi. 
That might have been changed, but I dont understand how to use it.

**Comment by rncbc** [comment]
what jackd version may I ask?

**Comment by henning** [comment]
Thanks for the fast response and sorry for not mentioning that info.

Actually it's harder to say then I thought:

dpkg -l says "5+nmu1"

and the result of "jackd --version" is:

jackdmp version 1.9.17 tmpdir /dev/shm protocol 8

About the driver:
Yes, I use seq
Actually so far this always worked pretty well for me. 
Especially with one specific device, it allows me to have it connected to a sequencer and DAW and at the same time running it's Midi editor, so I can modifiy the complex synth features while playing. 
With the plain Alsa driver The Port is blocked when using either of those connections, so I would have to stop playing and remove all connections to make adjustments to the synth( Novation Circuit that is, and it's Components UI in the chrome/chromium)

What's so wrong about it, what problems could I get?

**Comment by rncbc** [comment]
note that if any device is under exclusive control of any MIDI ports (eg. "raw" or yet some java OEM vendor editors out there) then all other MIDI interfaces are bonkers!

**Comment by henning** [comment]
I'm not sure if I understand that correctly and maybe my description wasnt explained well... The problem with blocking devices/ports appears for me only when I use "none" option for midi, so basically using plain alsa. 

With the seq Midi driver everything worked fine for me so far, and still is, besides that since 0.9.1 qjackctl  doesn't show me the midi port named (instead of numbered) aliases and all configuration settings having anything to do with port names and aliases didn't have any effect.

**Comment by rncbc** [comment]
does is also show the same "just numbers" port names in the Connections window?

you can try (re)moving away the (old) `~/.config/rncbc.org/QjackCtl.conf`, while qjackctl is NOT running, and start all over from scratch... and check whether those ports still show the numbers instead of the complete names.

**Comment by henning** [comment]
nope. Stoped jack and qjackctl, moved the config, started qjackctl, re-entered the basic audio interface settings, started jack.
Jack midi ports are only "midi_capture_1" etc instead of "Circuit:midi/playback_1" as ist was before the update of UBuntu and with it jackd and qjackctl. Jack_lsp -A is showing all those aliases perfectly, the python script that I use to save and restore connections works.

**Comment by rncbc** [comment]
i'm out of ideas. "midi_capture_N" are, as ever been, the native names for the jack-midi system input ports. on my system, for instance, they get also the "human readable" as you say, pretty-names as you say: however these are not aliases, they are set in the JACK **meta-data** database; maybe you're having some stalled db?

**Comment by henning** [comment]
I use the term aliases because when running jack_lsp I see those names when I add the option "-A" which is documented as aliases.

I have never set anything myself in any jack meta data system.
To me they appear the same or very similar as the names that the devices have in the alsa midi system. 

So if there is any db, I dont know where it is and how it has been filled with something, or how I can reset it.
Do you have any hint?

**Comment by rncbc** [comment]
hint: `/dev/shm/jack_db` seems to be the place here; make sure you kill all `jackd` and `jackdbus` before you do something in that regard.

**Comment by henning** [comment]
Well, I dont have that file:

```
$ sudo ls /dev/shm/*jack*
 /dev/shm/jack_sem.1000_default_Carla		    /dev/shm/jack_sem.1000_default_ModelSamples		  '/dev/shm/jack_sem.1000_default_PulseAudio JACK Source'
'/dev/shm/jack_sem.1000_default_Jacker MIDI Tool'  '/dev/shm/jack_sem.1000_default_PulseAudio JACK Sink'   /dev/shm/jack-shm-registry
```

**Comment by henning** [comment]
I now further tried to install older versions of qjackctl to find which one has been working, but I couldnt find any... there must be something in the way the latest jack in Ubuntu and qjacktctl work together that makes it fail.

Anyway, what I realized: 

in the version 0.5.0 the "normal" connections window seems to work better than in more recent versions (see below for description how it fails).

In 0.5.0 It display the port aliases when selected "first" in "display ->  Jack client/port aliases" (you see, you also call these things aliases), and then the names are shown, and connections can be made and are displayed.

On all newer version that I have tried - 0.6.3, 0.9.0,. 0.9.4 when trying ti use the connect view to connect devices, I can drag and drop a device name from the left to another device on the right, but the line displaying the connection only blinks very short, and then disappears.

The connections than is totally fine working and is displayed on the commend line, or with my own python script, but in qjacjctl it's invisible.

What could be the reason for this and how could it possibly be fixed?

**Comment by henning** [comment]
I now further tried to install older versions of qjackctl to find which one has been working, but I couldnt find any... there must be something in the way the latest jack in Ubuntu and qjacktctl work together that makes it fail.

Anyway, what I realized: 

in the version 0.5.0 the "normal" connections window seems to work better than in more recent versions (see below for description how it fails).

In 0.5.0 It display the port aliases when selected "first" in "display ->  Jack client/port aliases" (you see, you also call these things aliases), and then the names are shown, and connections can be made and are displayed.

On all newer version that I have tried - 0.6.3, 0.9.0,. 0.9.4 when trying ti use the connect view to connect devices, I can drag and drop a device name from the left to another device on the right, but the line displaying the connection only blinks very short, and then disappears.

The connections than is totally fine working and is displayed on the commend line, or with my own python script, but in qjacjctl it's invisible.

What could be the reason for this and how could it possibly be fixed?

ss
